substrate/frame/contracts/src/benchmarking/mod.rs

```diff
@@ -910,7 +910,7 @@ benchmarks! {
 					value: code_hashes_bytes,
 				},
 			],
-			deploy_body: Some(body::repeated_dyn(r, vec![
+			deploy_body: Some(body::repeated_dyn(code_hash_len as _, vec![
```

Comment:
Has to be repeated for the number of code_hashes not the length of each individual code_hash. Please add some assertion in the setup phase that the contract actually has these number of dependencies.

```suggestion
			deploy_body: Some(body::repeated_dyn(T::MaxDelegateDependencies::get(), vec![
```